practice_2/GreatDayApp/Extensions/StringExtensions.swift

```diff
@@ -0,0 +1,7 @@
+import Foundation
+
+extension String {
+    func matches(_ regex: String) -> Bool {
+        return self.range(of: regex, options: .regularExpression, range: nil, locale: nil) != nil
```

Comment:
try do not compare to nil if possible

Reply:
If I do it like below, I get a swiftlint warning: Unused Optional Binding Violation: Prefer `!= nil` over `let _ =` 
This is the reason why I did it this way. Perhaps there are still ways to handle this point. I would be glad if you tell me. Thanks.
`
    func matches(_ regex: String) -> Bool {

        if let _ = self.range(of: regex, options: .regularExpression, range: nil, locale: nil) {
            return true
        }
        return false
    }
`

Reply:
LGTM, leave it as is
but some thoughts are here:
https://forums.swift.org/t/a-more-swifty-way-to-check-an-optional-for-a-nil-value/19221